app/src/main/java/app/nexd/android/ui/seeker/create/address/SeekerCreateRequestConfirmAddressFragment.kt

```diff
@@ -0,0 +1,73 @@
+package app.nexd.android.ui.seeker.create.address
+
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.activity.addCallback
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import androidx.navigation.fragment.findNavController
+import app.nexd.android.R
+import app.nexd.android.databinding.FragmentSeekerCreateRequestConfirmAddressBinding
+import app.nexd.android.di.sharedGraphViewModel
+import app.nexd.android.ui.seeker.create.SeekerCreateRequestViewModel
+import kotlinx.android.synthetic.main.fragment_seeker_create_request_confirm_address.*
+
+class SeekerCreateRequestConfirmAddressFragment : Fragment() {
+    private val vm: SeekerCreateRequestViewModel by sharedGraphViewModel(R.id.nav_seeker_create_request)
+
+    private lateinit var binding: FragmentSeekerCreateRequestConfirmAddressBinding
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        binding =
+            FragmentSeekerCreateRequestConfirmAddressBinding.inflate(inflater, container, false)
+        binding.viewModel = vm
+        binding.lifecycleOwner = viewLifecycleOwner
+        return binding.root
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val onBackPressedCallback = requireActivity().onBackPressedDispatcher.addCallback(this) {
+            vm.setStateToLoading()
+        }
```

Comment:
I added a onBackPressedCallBack in order to handle the situation in which the user has selected amount of articles and pressed ok (>> moved to address confirmation) checks his address but presses back to review the articles.

The problem that arose was that the ViewModel state was on PROCESSING and the fragment in which the user can select the articles has a listener to the viewModel state.
If the ViewModel state is PROCESSING the user is automatically navigated to the address confirmation screen.

Thus, I wanted to change the status of the viewModel in case the user pressed back, so he won't be navigated instantly to the address confirmation screen again.

But as changing the status of the VM from the fragment is already a violation of the design pattern, I'll try to come up with a different solution
(open to any suggestions! :) )